c++/src/capnp/rpc-twoparty.c++

```diff
@@ -192,7 +192,7 @@ public:
           messages[i].fds = ownMessages[i]->fds;
         }
         return network.getStream().writeMessages(messages).attach(kj::mv(ownMessages), kj::mv(messages));
-      }).catch_([this](kj::Exception&& e) {
+      }).catch_([&network](kj::Exception&& e) {
```

Comment:
I see the `networker.queuedMessages.add(kj::addRef(*this))` on like 173 above... what I'm wondering is whether the `kj::mv(network.queuedMessages)` on line 186 could be problematic for this catch? I don't suspect so but is there any chance the captured `&network` on this catch handler could end up invalid after that move on 186? Just making sure I have the mental model correct.

Reply:
If the `network` had been destroyed, then its field `previousWrite` would have been destroyed, thereby cancelling this promise. Thus, we can conclude `network` is definitely still valid in this lambda.

Reply:
Got it!